csharp/src/Ice/Reference.cs

```diff
@@ -238,892 +338,512 @@ public override string ToString()
-                // the adapter id string in quotes.
-                //
-                string a = StringUtil.EscapeString(_adapterId, null, Communicator.ToStringMode);
-                if (StringUtil.FindFirstOf(a, " :@") != -1)
-                {
-                    s.Append('"');
-                    s.Append(a);
-                    s.Append('"');
-                }
-                else
-                {
-                    s.Append(a);
-                }
+                ostr.WriteString(AdapterId);
             }
-            return s.ToString();
         }
 
-        public override Dictionary<string, string> ToProperty(string prefix)
+        internal Dictionary<string, string> ToProperty(string prefix)
         {
+            if (IsFixed)
+            {
+                throw new NotSupportedException("cannot convert a fixed proxy to a property dictionary");
+            }
+
             var properties = new Dictionary<string, string>
             {
                 [prefix] = ToString(),
-                [prefix + ".CollocationOptimized"] = _collocationOptimized ? "1" : "0",
+                [prefix + ".CollocationOptimized"] = IsCollocationOptimized ? "1" : "0",
                 [prefix + ".ConnectionCached"] = IsConnectionCached ? "1" : "0",
-                [prefix + ".PreferSecure"] = _preferSecure ? "1" : "0",
+                [prefix + ".PreferSecure"] = IsPreferSecure ? "1" : "0",
                 [prefix + ".EndpointSelection"] =
-                       _endpointSelection == EndpointSelectionType.Random ? "Random" : "Ordered",
-                [prefix + ".LocatorCacheTimeout"] = _locatorCacheTimeout.ToString(CultureInfo.InvariantCulture),
+                       EndpointSelection == EndpointSelectionType.Random ? "Random" : "Ordered",
+                [prefix + ".LocatorCacheTimeout"] = LocatorCacheTimeout.ToString(CultureInfo.InvariantCulture),
                 [prefix + ".InvocationTimeout"] = InvocationTimeout.ToString(CultureInfo.InvariantCulture)
             };
 
-            if (_routerInfo != null)
+            if (RouterInfo != null)
             {
-                Dictionary<string, string> routerProperties = _routerInfo.Router.IceReference.ToProperty(prefix + ".Router");
+                Dictionary<string, string> routerProperties =
+                    RouterInfo.Router.IceReference.ToProperty(prefix + ".Router");
                 foreach (KeyValuePair<string, string> entry in routerProperties)
                 {
                     properties[entry.Key] = entry.Value;
                 }
             }
-
-            if (_locatorInfo != null)
+            if (LocatorInfo != null)
             {
-                Dictionary<string, string> locatorProperties = _locatorInfo.Locator.IceReference.ToProperty(prefix + ".Locator");
+                Dictionary<string, string> locatorProperties =
+                    LocatorInfo.Locator.IceReference.ToProperty(prefix + ".Locator");
```

Comment:
Shouldn't this be `LocatorInfo.Locator.ToProperty(prefix + ".Locator")`?

Reply:
Fixed, it's really the same since IObjectPrx.ToProperty called ToProperty on the reference.